If the user types anything other than an existing file the program will throw an exception.
We can elegantly fix this using  the **try/except** structure.
```
fname = input('Enter the file name: ')
try:
    fhand = open(fname)
except:
    print(File cannot be opened: ', fname)
    exit()
count = 0
for line in fhand:
    if line.startswith('Subject: '):
        count += 1
print('There were', count, 'subject lines in', fname)
```

Now if the user enters anything other than an existing file our try/except block catches them gracefully
```
python search7.py
Enter the file name: mbox.txt
There were 1797 subject lines in mbox.txt
python search7.py
Enter the file name: na na boo boo
File cannot be opened: na na boo boo
```